source/js/foundation/pages/dear-internet/template-js-handler/plane.js

```diff
@@ -0,0 +1,95 @@
+const SVG_WRAPPER = document.querySelector("#plane-pathway-wrapper");
+const SVG = document.querySelector("#plane-pathway-wrapper svg");
+const PATH = document.querySelector("#plane-pathway-wrapper #path");
+const PLANE = document.querySelector("#plane-pathway-wrapper #plane");
+
+class Plane {
+  constructor() {
+    this.defaultPlaneRotation = 0;
+    this.init();
+  }
+
+  movePlane() {
+    if (SVG_WRAPPER.classList.contains("invisible")) {
+      SVG_WRAPPER.classList.remove("invisible");
+    }
+
+    // distance scrolled
+    let scrollDist =
+      document.documentElement.scrollTop + document.body.scrollTop;
+
+    // what percentage has the user scrolled?
+    let scrollPercentage = scrollDist / SVG.scrollHeight;
+
+    // find path length
+    let pathLen = PATH.getTotalLength();
+
+    // get the position of a point at <scrollPercentage> along the path.
+    let point = PATH.getPointAtLength(scrollPercentage * pathLen);
+
+    // get the position of a point slightly ahead so we can calculate the angle we want the plane to rotate
+    let pointAhead = PATH.getPointAtLength(
+      (scrollPercentage + 0.0001) * pathLen
+    );
+
+    // find rotating angle
+    let angleDeg =
+      Math.atan2(pointAhead.y - point.y, pointAhead.x - point.x) *
+      (180 / Math.PI);
+
+    // position plane at this point
+    PLANE.setAttribute(
+      "transform",
+      `translate(${point.x}, ${point.y}) rotate(${
+        angleDeg + this.defaultPlaneRotation
+      } 0 0)`
+    );
+
+    this.revealPath(scrollPercentage);
+  }
+
+  revealPath(scrollPercentage) {
+    // revealing/hiding PATH by adjusting the dash gap size
+    PATH.setAttribute(
+      "stroke-dashoffset",
+      (1 - scrollPercentage) * PATH.getTotalLength()
+    );
+  }
+
+  findDefaultPlaneRotation() {
+    if (PLANE.getAttribute("transform")) {
+      // example of "transform" attribute value:
+      // "translate(num num num) rotate(num num num)"
+      let regex = /\brotate\b\([-.0-9]+\s[-.0-9]+\s[-.0-9]+\)/g;
+
+      if (PLANE.getAttribute("transform").match(regex)) {
+        this.defaultPlaneRotation = PLANE.getAttribute("transform")
+          .match(regex)[0]
+          .replace("rotate", "")
+          .replace("(", "")
+          .replace(")", "")
+          .split(" ")[0]
+          .trim();
+      }
+      this.defaultPlaneRotation = parseFloat(this.defaultPlaneRotation);
+    }
```

Comment:
We can use a regex capture group here:

```suggestion
    if (PLANE.getAttribute("transform")) {
      // Get the first rotation value out of the transforms.
      // Example of "transform" attribute value:
      // "translate(num num num) rotate(num num num)"
      const match = PLANE.getAttribute("transform").match(/rotate\((\S+)[ )]/)[1];
      if (match) {
        this.defaultPlaneRotation = parseFloat(match[1]);
      }
    }
```

If we only care about the first rotation argument, we can actually look for "not whitespace" after the opening parenthesis, and then if that exists, parse it as number.

```
rotate   - just the word
\(       - literal (
(\S+)    - one or more "not whitespace" characters, captured as a group
[ )]     - either a space, or literal )
```

Reply:
Ahhhh you are right! Thanks for the explanation too. Totally forgot that we can grab the first argument directly from regex too lol 🙈